test/data_map_test.exs

```diff
@@ -0,0 +1,162 @@
+defmodule DataMapTest do
```

Comment:
I think for now it's not important, but I would have put public all the validators and test them individually